mailpoet/lib/Mailer/MailerLog.php

```diff
@@ -172,6 +178,34 @@ public static function processError(
     self::enforceExecutionRequirements();
   }
 
+  /**
+   * Process error, increase transactional_email_error_count and pauses sending if it reaches retry limit
+   * This method is meant to be used for processing errors when sending transactional emails
+   * like: Confirmation Email, Preview email, Stats Notification etc.
+   *
+   * @throws \Exception
+   */
+  public static function processTransactionalEmailError(
+    string $operation,
+    string $errorMessage,
+    ?string $errorCode = null
+  ): void {
+    $mailerLog = self::getMailerLog();
+    $lastErrorTime = $mailerLog['transactional_email_last_error_at'] ?? null;
+    $ignoreErrorThreshold = time() - (2 * 60); // 2 minutes ago
+    // We want to log the error max one time per 2 minutes
+    if ($lastErrorTime && $lastErrorTime > $ignoreErrorThreshold) {
+      return;
+    }
+    $mailerLog = self::setError($mailerLog, $operation, $errorMessage, $errorCode);
+    $mailerLog['transactional_email_last_error_at'] = time();
+    $mailerLog['transactional_email_error_count'] = ($mailerLog['transactional_email_error_count'] ?? 0) + 1;
+    self::updateMailerLog($mailerLog);
+    if ($mailerLog['transactional_email_error_count'] > self::RETRY_ATTEMPTS_LIMIT) {
```

Comment:
> When confirmation email sending fails three times (with at least 2 minutes interval in between those failures), the site admin should see a notice on the emails page.

Since we are using `>` here instead of `>=`, we are pausing sending and showing the admin notice when the confirmation email sending fails four times instead of three times. This is just a detail, and I think either way works, I'm mentioning it just for consistency. If we decide to keep it the way it is, we might want to update the PR and the ticket descriptions.

Reply:
@rodrigoprimo Nice catch! I'll update the condition.